### PR TITLE
Feat/appl records/v77

### PR DESCRIPTION
**Pull request by victorjulien**
#6756 

with rust api cleanups
removal of useless c telnet files
actual disabling of frames logging

suricata-verify-pr: 631

### CODE REVIEW

File: rust/src/frames.rs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+pub fn frame_set_tx(flow: *const Flow, dir: i32, frame_id: i64, tx_id: u64)
+{
+    unsafe { AppLayerFrameSetTxIdById(flow, dir, frame_id, tx_id) }
+}
+
+pub fn frame_add_event(flow: *const Flow, dir: i32, frame_id: i64, event: u8)
+{
+    unsafe { AppLayerFrameAddEventById(flow, dir, frame_id, event) }
+}
```
Comment (jasonish) [inline rust/src/frames.rs:61]: These look like they are unused now?

**Comment by jasonish** [comment]
Payload and packet logging is optional for alerts, should logging the frame be optional and disabled by default as well to match that behaviour?

**Comment by jasonish** [comment]
The telnet parser had me thinking... What would a protocol logger for telnet look like?

**Comment by suricata-qa** [comment]
ERROR: QA failed on tlpr1_asan_suri.
  

Pipeline 5603

**Comment by victorjulien** [comment]
> Payload and packet logging is optional for alerts, should logging the frame be optional and disabled by default as well to match that behaviour?

I think yes, but I'm wondering if we should distinguish between frame and frame payload.

Examples

Full:
```json
{
  "timestamp": "1999-11-28T02:12:58.716445+0000",
  "flow_id": 954228178086019,
  "pcap_cnt": 56,
  "event_type": "alert",
  "src_ip": "192.168.0.2",
  "src_port": 1550,
  "dest_ip": "192.168.0.1",
  "dest_port": 23,
  "proto": "TCP",
  "alert": {
    "action": "allowed",
    "gid": 1,
    "signature_id": 1,
    "rev": 0,
    "signature": "",
    "category": "",
    "severity": 3
  },
  "app_proto": "telnet",
  "flow": {
    "pkts_toserver": 26,
    "pkts_toclient": 27,
    "bytes_toserver": 1991,
    "bytes_toclient": 2547,
    "start": "1999-11-28T02:12:38.387203+0000"
  },
  "frame": {
    "type": "data",
    "id": 72,
    "offset": 215,
    "length": 26,
    "direction": "toserver",
    "complete": true,
    "payload": "L3NiaW4vcGluZyB3d3cueWFob28uY29tDQo=",
    "payload_printable": "/sbin/ping www.yahoo.com\r\n",
  }
}
```
No payload
```json
{
  "timestamp": "1999-11-28T02:12:58.716445+0000",
  "flow_id": 954228178086019,
  "pcap_cnt": 56,
  "event_type": "alert",
  "src_ip": "192.168.0.2",
  "src_port": 1550,
  "dest_ip": "192.168.0.1",
  "dest_port": 23,
  "proto": "TCP",
  "alert": {
    "action": "allowed",
    "gid": 1,
    "signature_id": 1,
    "rev": 0,
    "signature": "",
    "category": "",
    "severity": 3
  },
  "app_proto": "telnet",
  "flow": {
    "pkts_toserver": 26,
    "pkts_toclient": 27,
    "bytes_toserver": 1991,
    "bytes_toclient": 2547,
    "start": "1999-11-28T02:12:38.387203+0000"
  },
  "frame": {
    "type": "data",
    "id": 72,
    "offset": 215,
    "length": 26,
    "direction": "toserver",
    "complete": true,
  }
}
```
No frame
```json
{
  "timestamp": "1999-11-28T02:12:58.716445+0000",
  "flow_id": 954228178086019,
  "pcap_cnt": 56,
  "event_type": "alert",
  "src_ip": "192.168.0.2",
  "src_port": 1550,
  "dest_ip": "192.168.0.1",
  "dest_port": 23,
  "proto": "TCP",
  "alert": {
    "action": "allowed",
    "gid": 1,
    "signature_id": 1,
    "rev": 0,
    "signature": "",
    "category": "",
    "severity": 3
  },
  "app_proto": "telnet",
  "flow": {
    "pkts_toserver": 26,
    "pkts_toclient": 27,
    "bytes_toserver": 1991,
    "bytes_toclient": 2547,
    "start": "1999-11-28T02:12:38.387203+0000"
  },
}
```

**Comment by victorjulien** [comment]
replaced by #6758

**Pull request closed without merging**